test/clj_puppetdb/http_test.clj

```diff
@@ -56,53 +56,49 @@
 
     (testing "Should have proper response"
       (with-redefs [http/request-with-client (fn [_ _ _] (future (fake-get 200)))]
-        (let [GET-response (GET client path params)]
+        (let [GET-response (GET client q-path q-params)]
           (is (= (first GET-response) response-data))
           (is (= (second GET-response) response-headers)))))
 
     (testing "Should throw proper exception"
       (with-redefs [http/request-with-client (fn [_ _ _] (future (fake-get 400)))]
-        (try
-          (GET client path params)
-          (catch ExceptionInfo ei
-            (let [info (.getData ei)]
-              (is (= (:status info) 400))
-              (is (= (:kind info) :puppetdb-query-error))
-              (is (= (:params info) params))
-              (is (= (:endpoint info) path))
-              (is (= (:host info) host))
-              (is (= (:msg info) response-data-encoded)))))))
+        (try+
+          (GET client q-path q-params)
+          (catch [] {:keys [status kind params endpoint host msg]}
+            (is (= status 400))
+            (is (= kind :puppetdb-query-error))
+            (is (= params q-params))
+            (is (= endpoint q-path))
+            (is (= host q-host))
+            (is (= msg response-data-encoded))))))
```

Comment:
Another way we could do this if we wanted to would be

```clojure
  (is (thrown+? [:kind :puppetdb-query-error
                       :params q-params
                       :endpoint q-path
                       :host q-host
                       :status 400
                       :msg response-data-encoded]
                      (GET client q-path q-params)))
```

I'm not sure what I think about this, but just wanted to bring it up as an idea.